tests/src/test/java/io/streamnative/pulsar/handlers/kop/SaslPlainTest.java

```diff
@@ -193,4 +197,24 @@ void badNamespaceProvided() throws Exception {
             assertTrue(e.getMessage().contains("SaslAuthenticationException"));
         }
     }
+
+    @Test
+    void clientWithoutAuth() throws Exception {
+        final int metadataTimeoutMs = 8000;
+
+        Properties props = new Properties();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:" + getKafkaBrokerPort());
+        props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+        props.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, Integer.toString(metadataTimeoutMs));
+
+        KafkaProducer<String, String> producer = new KafkaProducer<>(props);
+        producer.send(new ProducerRecord<>(TOPIC, "", "hello"), (recordMetadata, e) -> {
+            assertNotNull(e);
+            assertTrue(e.getMessage().contains("Failed to update metadata"));
+        });
+
+        Thread.sleep(metadataTimeoutMs + 300);
```

Comment:
why it need a sleep here?

Reply:
Oh, I see, the assert is in the callback.

Reply:
Do we have a better way to avoid sleep? sleep will cause a flaky test, because the ci env is not stable.